### DISCUSSION TITLE
Printhead settings differ from Cura's official SM A350 profile

### DISCUSSION THREAD

**Discussion by Wicklunds** [Kaouthia/Snapmaker-2#4] [Q&A] [answered]
I installed Cura 4.9 today, and noticed that the Printer profile list has a "Snapmaker 2 A350" entry!

Most of the settings are the same as shown in your profile, however the Printhead settings differ.

Yours:
![](https://i.imgur.com/kuFcgpX.png)

Theirs:
![](https://i.imgur.com/uxSu68P.png)

So two questions:
1) Do you have an idea of why their values are different?
2) Will there be an issue if I stick with their (slightly different) profile settings?

Thanks!

**Comment by Wicklunds** [comment]
I also notice that their end code is slightly different.

Yours: (differing lines in bold)
> [...]
> G1 E-1 F300 ;retract the filament a bit before lifting the nozzle, to release some of the pressure
> **G0 Z330** ;move Z axis up to max height
> **G28 X0 Y0** ;home the print head

Theirs: (differing lines in bold)
> [...]
> G1 E-1 F300 ;retract the filament a bit before lifting the nozzle, to release some of the pressure
> **G1 Z{machine_width} E-1 F3000** ;move Z up a bit and retract filament even more
> **G1 X0 F3000** ;move X to min endstops, so the head is out of the way
> **G1 Y{machine_depth} F3000** ;so the head is out of the way and Plate is moved forward

So a third question:
3) Are the end-g-code differences an issue?

(For example, I notice that their end g-code says "Z{machine_**width**}", whereas I would expect it to be "Z{machine_**height**}", given the comment of "move Z up a bit".)

**Comment by Kaouthia** [comment, marked as answer]
Their printhead values are different because they have access to the actual CAD models. I was simply getting it "close enough" with a tape measure. So theirs will be more physically accurate to the printer, but mine just give an extra couple of mm of clearance is all (I measured a little extra than needed to be safe). It's purely to stop the head bumping into other parts when you print models in sequence (which most people won't need to do that often, if ever, really).

As for the end gcode settings, the only difference really is the how we move things. They move the Z up, then the X across, then the Y forward in separate operations. Mine moves the Z up and then brings the X & Y to zero at the same time to save a second or two of time. That's all, really.

It's strange that they're using "{machine_width}" rather than "{machine_height}" for the Z axis. Personally, I just dialled the height in manually. A350 printers will go to the 330mm (the max height it can print), A250 and A150 printers will go to whatever their max height is and stop because they can't go any higher. But I probably should modify mine to use {machine_height}.

**Reply by Wicklunds** [reply]
Thanks for the details! It sounds like either profile should work then.

One last question: Your TPU profile appears to have "retraction" enabled.

This surprised me, because many people in this Snapmaker Forums thread says you should disable retraction for TPU prints: https://forum.snapmaker.com/t/printing-tpu-with-a250-or-any-sm-2-0/6217

Is there a reason your TPU profile did not require retraction to be disabled?

Or should I go ahead and disable retraction, to increase the chance of a successful print?

**Reply by Kaouthia** [reply]
The reason is because I didn't feel it was necessary. It's only retracting by 0.1mm, which is a ludicrously tiny amount, just to help stop it from stringing. It hasn't caused me any issues, but not all TPU is the same. With some it'll be fine, and with others you'll want it off completely.

So far, for me, I've yet to experience any kind of TPU filament where 0.1mm of retraction was an issue. The reason most people say to turn it off because they have the setting way too high at like 7-8mm. And when it's that high it WILL cause issues.

**Reply by Wicklunds** [reply]
Good to hear; and thanks for explaining with details! It definitely helps newcomers learn faster. :)

I'll go ahead and try with retraction enabled, and see how it goes. (if I hit issues, I'll give an update in this thread for other readers)
